src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorPicker.cs

```diff
@@ -0,0 +1,1434 @@
+using Windows.Foundation;
+using Windows.UI;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+using Windows.UI.Xaml.Shapes;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	public partial class ColorPicker : Control
+	{
+		private enum ColorUpdateReason
+		{
+			InitializingColor,
+			ColorPropertyChanged,
+			ColorSpectrumColorChanged,
+			ThirdDimensionSliderChanged,
+			AlphaSliderChanged,
+			RgbTextBoxChanged,
+			HsvTextBoxChanged,
+			AlphaTextBoxChanged,
+			HexTextBoxChanged,
+		};
+
+		private bool m_updatingColor = false;
+		private bool m_updatingControls = false;
+		private Rgb m_currentRgb = new Rgb(1.0, 1.0, 1.0);
+		private Hsv m_currentHsv = new Hsv(0.0, 1.0, 1.0);
+		private string m_currentHex = "#FFFFFFFF";
+		private double m_currentAlpha = 1.0;
+
+		private string m_previousString = string.Empty;
+		private bool m_isFocusedTextBoxValid = false;
+
+		private bool m_textEntryGridOpened = false;
+
+		// Template parts
+		private Primitives.ColorSpectrum m_colorSpectrum;
+
+		private Grid m_colorPreviewRectangleGrid;
+		private Rectangle m_colorPreviewRectangle;
+		private Rectangle m_previousColorRectangle;
+		private ImageBrush m_colorPreviewRectangleCheckeredBackgroundImageBrush;
+
+		private IAsyncAction m_createColorPreviewRectangleCheckeredBackgroundBitmapAction = null;
+
+		private Primitives.ColorPickerSlider m_thirdDimensionSlider;
+		private LinearGradientBrush m_thirdDimensionSliderGradientBrush;
+
+		private Primitives.ColorPickerSlider m_alphaSlider;
+		private LinearGradientBrush m_alphaSliderGradientBrush;
+		private Rectangle m_alphaSliderBackgroundRectangle;
+		private ImageBrush m_alphaSliderCheckeredBackgroundImageBrush;
+
+		private IAsyncAction m_alphaSliderCheckeredBackgroundBitmapAction = null;
+
+		private ButtonBase m_moreButton;
+		private TextBlock m_moreButtonLabel;
+    
+		private ComboBox m_colorRepresentationComboBox;
+		private TextBox m_redTextBox;
+		private TextBox m_greenTextBox;
+		private TextBox m_blueTextBox;
+		private TextBox m_hueTextBox;
+		private TextBox m_saturationTextBox;
+		private TextBox m_valueTextBox;
+		private TextBox m_alphaTextBox;
+		private TextBox m_hexTextBox;
+
+		private ComboBoxItem m_RgbComboBoxItem;
+		private ComboBoxItem m_HsvComboBoxItem;
+		private TextBlock m_redLabel;
+		private TextBlock m_greenLabel;
+		private TextBlock m_blueLabel;
+		private TextBlock m_hueLabel;
+		private TextBlock m_saturationLabel;
+		private TextBlock m_valueLabel;
+		private TextBlock m_alphaLabel;
+
+		private SolidColorBrush m_checkerColorBrush;
+
+		// Uno Doc: Added to dispose event handlers
+		private SerialDisposable _eventSubscriptions = new SerialDisposable();
+
+		public ColorPicker()
+		{
+			// Uno Doc: Not supported
+			//__RP_Marker_ClassById(RuntimeProfiler::ProfId_ColorPicker);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)